src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/LinearAVF.java

```diff
@@ -292,360 +192,49 @@ public double getWeight(Criterion crit) {
-   *
-   * @param value >= 0
-   */
-  private void setNbMinNightSubjectiveValueWeight(double value) {
-    this.nbMinNightSubjectiveValueWeight = value;
-    LOGGER.debug("The number of minimum night weight has been set to {}", value);
-  }
-
-  /**
-   * Set the weight of the television subjective value corresponding to the importance of the
-   * television criteria.
-   *
-   * @param value >= 0
-   */
-  private void setTeleSubjectiveValueWeight(double value) {
-    this.teleSubjectiveValueWeight = value;
-    LOGGER.debug("The tele weight has been set to {}", value);
-  }
-
-  /**
-   * Gets the object PartialValueFunction used to compute the subjective value of the floor Area
-   *
-   * @return the attribute floorAreaValueFunction
-   */
-  public LinearValueFunction getFloorAreaValueFunction() {
-    return this.floorAreaValueFunction;
-  }
-
-  /**
-   * Set the function which will be used to calculate the subjective value of the floor area.
-   *
-   * @param floorAreaValueFunction the new LinearValueFunction for FloorArea
-   */
-  private void setFloorAreaValueFunction(LinearValueFunction floorAreaValueFunction) {
-    this.floorAreaValueFunction = checkNotNull(floorAreaValueFunction);
-    LOGGER.info("The floor area preferencies has been set");
-  }
-
-  /**
-   * Gets the object PartialValueFunction used to compute the subjective value of the floor Area
-   * Terrace
-   *
-   * @return the attribute floorAreaTerraceValueFunction
-   */
-  public LinearValueFunction getFloorAreaTerraceValueFunction() {
-    return this.floorAreaTerraceValueFunction;
-  }
-
-  /**
-   * Set the function which will be used to calculate the subjective value of the floor area of an
-   * existing terrace.
-   *
-   * @param floorAreaTerraceValueFunction the new LinearValueFunction for FloorAreaTerrace
-   */
-  private void setFloorAreaTerraceValueFunction(LinearValueFunction floorAreaTerraceValueFunction) {
-    this.floorAreaTerraceValueFunction = checkNotNull(floorAreaTerraceValueFunction);
-    LOGGER.info("The floor area of the terrace preferencies has been set");
-  }
-
-  /**
-   * Gets the object PartialValueFunction used to compute the subjective value of the number of
-   * bathrooms
-   *
-   * @return the attribute nbBathroomsValueFunction
-   */
-  public LinearValueFunction getNbBathroomsValueFunction() {
-    return this.nbBathroomsValueFunction;
-  }
-
-  /**
-   * Set the function which will be used to calculate the subjective value of the number of
-   * bathrooms.
-   *
-   * @param nbBathroomsValueFunction the new LinearValueFunction for NbBathrooms
-   */
-  private void setNbBathroomsValueFunction(LinearValueFunction nbBathroomsValueFunction) {
-    this.nbBathroomsValueFunction = checkNotNull(nbBathroomsValueFunction);
-    LOGGER.info("The number of bathrooms preferencies has been set");
+  private void setInternalValueFunction(Criterion criterion, BooleanValueFunction vf) {
+    checkArgument(criterion.hasBooleanDomain());
```

Comment:
Peut-être faire un checkNotNull sur BooleanValueFunction

Reply:
Fix in d34a55ad54fb6e348608d341481c740cbb7ff702